pilopt/src/referenced_symbols.rs

```diff
@@ -149,8 +167,226 @@ fn symbols_in_expression(
     }
 }
 
-impl ReferencedSymbols for Type {
+fn symbols_in_expression_asm(
```

Comment:
should this be a `symbols()` method in `ExpressionAsm` objects? maybe implement the `ReferencedSymbols` trait?

Reply:
It's also implemented. But since symbols() uses all_children this function is used to get “first level” PublicReferences/References or nothing, without checking their children.